extensions/amp-render/1.0/amp-render.js

```diff
@@ -380,7 +414,17 @@ export class AmpRender extends BaseElement {
                         bind.signals().get('FIRST_MUTATE') === null
                       ),
                     })
-                    .then(() => dict({'__html': element./* OK */ innerHTML}));
+                    .then(() =>
+                      dict({
+                        // We should use innerHTML when the template lacks a wrapper
+                        // element, outerHTML otherwise in order to include the wrapper
+                        // element itself.
+                        '__html':
+                          nonEmptyNodeCount === 1
+                            ? element./* OK */ outerHTML
+                            : element./* OK */ innerHTML,
```

Comment:
We may have missed an edge case. A template like the following has a single `TextNode`:

```html
<template>
  {{{foo}}}
</template>
```

with `{ "foo": "<p>hola</p>" }`, we'd be inserting the superfluous outer `<div>`:

```html
<div><p>hola</p></div>
```

I think we should keep `nonEmptyNodes` instead of a `nonEmptyNodeCount`, and change our condition for `outerHTML`

```suggestion
                          (
                            nonEmptyNodes.length === 1 &&
                            nonEmptyNodes[0].nodeType === Node. ELEMENT_NODE
                          )
                            ? element./* OK */ outerHTML
                            : element./* OK */ innerHTML,
```

Reply:
This is not the case. See below screenshot. I also updated the unit test to verify this case. 

<img width="1275" alt="Screen Shot 2021-08-02 at 9 14 17 PM" src="https://user-images.githubusercontent.com/506183/127958715-b3128d71-cc8e-4dac-a614-2f9a32c3ec3a.png">

Reply:
Oh nice!